frontend/src/metabase/visualizations/components/ChartWithLegend.jsx

```diff
@@ -21,6 +21,15 @@ class ChartWithLegend extends Component {
     style: {},
   };
 
+  componentDidUpdate(prevProps) {
```

Comment:
I believe a more suitable place for this would be right in `ExplicitSize` component:
In `__updateSize` we can change the line 186 to `this.setState({ width, height }, () => this.props.onResize?.());`

And then slightly tweak the prop you added in the PieChart: `onResize={this.updateChartViewportSize}`

Reply:
Thank you, Sasha!